Denis_Lozovy_1_HW.cpp

```diff
@@ -0,0 +1,61 @@
+#include <cmath>
+#include <iostream>
+
+int main() {
+
+  int a{};
+  int b{};
+  int c{};
+  constexpr int min{2};
+  constexpr int max{255};
+
+  std::cout << "Please enter a in range [" << min << ".." << max << "]\n";
+  std::cin >> a;
+  if (a < min or a > max) {
+    std::cout << "Incorrect a\n";
+    return -1;
+  }
+
+  std::cout << "\nPlease enter b in range [" << min << ".." << max << "]"
+            << std::endl;
+  std::cin >> b;
+  if (b < min or b > max) {
+    std::cout << "Incorrect b\n";
+    return -1;
+  }
+
+  std::cout << "\nPlease enter c in range [" << min << ".." << max << "]"
+            << std::endl;
+  std::cin >> c;
+  if (c < min or c > max) {
+    std::cout << "Incorrect c\n";
+    return -1;
+  }
+
+  // std::cout << a << " " << b << " " <<  c << "\n";
+  std::cout << "The equation is " << a << "x2 + " << b << "x + " << c
+            << " = 0\n";
+
+  long long D{(b * b) - (4 * a * c)};
+  long long discriminant{};
+  discriminant = std::sqrt(static_cast<__int64>(D));

```

Comment:
```suggestion
 const auto discriminant  { std::sqrt(D) };

```
нет смысла огбьявлять переменную а потом ее ниже инициализировать, можно это совместить.
Потом если у вас есть переменная которая ниже по коду не изменяет свое значение обьявляйте ее как const.
Вам не нужно преобразовывать функция sqrt вернет вещественное значение, т.е. переменная discriminant будет иметь тип double.
Ну и последенее а зачем эта переменная, вы же ее ниже по коду то не используете